python/cuspatial/cuspatial/_lib/linestring_distance.pyx

```diff
@@ -0,0 +1,39 @@
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+from cudf._lib.column cimport Column
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.column.column_view cimport column_view
+
+from cuspatial._lib.cpp.linestring_distance cimport (
+    pairwise_linestring_distance as cpp_pairwise_linestring_distance,
+)
+
+
+def pairwise_linestring_distance(
+    Column linestring1_offsets,
+    Column linestring1_points_x,
+    Column linestring1_points_y,
+    Column linestring2_offsets,
+    Column linestring2_points_x,
+    Column linestring2_points_y
+):
+    cdef column_view linestring1_offsets_view = linestring1_offsets.view()
+    cdef column_view linestring1_points_x_view = linestring1_points_x.view()
+    cdef column_view linestring1_points_y_view = linestring1_points_y.view()
+    cdef column_view linestring2_offsets_view = linestring2_offsets.view()
+    cdef column_view linestring2_points_x_view = linestring2_points_x.view()
+    cdef column_view linestring2_points_y_view = linestring2_points_y.view()
+
+    cdef unique_ptr[column] c_result
+    with nogil:
+        c_result = move(cpp_pairwise_linestring_distance(
```

Comment:
Why do you `move` here and in the return line below?

Reply:
I believe this relates to how cython translates it into cpp. Without `move` this line is tranlated into:
```cpp
        /* "cuspatial/_lib/linestring_distance.pyx":30
 *     cdef unique_ptr[column] c_result
 *     with nogil:
 *         c_result = cpp_pairwise_linestring_distance(             # <<<<<<<<<<<<<<
 *             linestring1_offsets_view,
 *             linestring1_points_x_view,
 */
        try {
          __pyx_t_2 = cuspatial::pairwise_linestring_distance(__pyx_v_linestring1_offsets_view, __pyx_v_linestring1_points_x_view, __pyx_v_linestring1_points_y_view, __pyx_v_linestring2_offsets_view, __pyx_v_linestring2_points_x_view, __pyx_v_linestring2_points_y_view);
        } catch(...) {
          #ifdef WITH_THREAD
          PyGILState_STATE __pyx_gilstate_save = __Pyx_PyGILState_Ensure();
          #endif
          __Pyx_CppExn2PyErr();
          #ifdef WITH_THREAD
          __Pyx_PyGILState_Release(__pyx_gilstate_save);
          #endif
          __PYX_ERR(0, 30, __pyx_L4_error)
        }
        __pyx_v_c_result = __pyx_t_2;
      }
```
The last line by default copies `__pyx_t_2` into a new variable and will error for `unique_ptr`.